doc/howto/etl.md

```diff
@@ -0,0 +1,348 @@
+```
+This command connects to the MySQL database using the specified connection string 
+and retrieves all the data from the "mytable" table. The data is stored in the 
+"data" variable as a Pandas DataFrame.
+
+**Note**: We can also use `%%sql df <<` to save the data into the `df` variable
+
+Since we'll be running locally via DuckDB we can simply Extract a public dataset and start working immediately.
+We're going to get our sample dataset (we will work with the Penguins datasets via a csv file):
+<!-- #endregion -->
+
+```python
+from urllib.request import urlretrieve
+
+_ = urlretrieve(
+    "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv",
+    "penguins.csv",
+)
+```
+
+And we can get a sample of the data to check we're connected and we can query the data:
+
+```sql
+SELECT *
+FROM penguins.csv
+LIMIT 3
+```
+
+### Transform data
+After extracting data, it's often necessary to transform it into a format that's 
+more suitable for analysis. This step may include cleaning data, filtering data, 
+aggregating data, and combining data from multiple sources. Here are some common 
+data transformation techniques:
+
+* **Cleaning data**: Data cleaning involves removing or fixing errors, inconsistencies, 
+   or missing values in the data. For example, you might remove rows with missing values, 
+   replace missing values with the mean or median value, or fix typos or formatting errors.
+* **Filtering data**: Data filtering involves selecting a subset of data that meets 
+   specific criteria. For example, you might filter data to only include records 
+   from a specific date range, or records that meet a certain threshold.
+* **Aggregating data**: Data aggregation involves summarizing data by calculating 
+   statistics such as the sum, mean, median, or count of a particular variable. 
+   For example, you might aggregate sales data by month or by product category.
+* **Combining data**: Data combination involves merging data from multiple sources 
+   to create a single dataset. For example, you might combine data from different 
+   tables in a relational database, or combine data from different files.
+
+In JupySQL, you can use Pandas DataFrame methods to perform data transformations or native SQL. 
+For example, you can use the rename method to rename columns, the dropna method to 
+remove missing values, and the astype method to convert data types. I'll demonstrate how to do it either with pandas or SQL.
+
+* Note: You can use either `%sql` or `%%sql`, check out the difference between the two [here](https://jupysql.ploomber.io/en/latest/community/developer-guide.html?highlight=%25sql%20vs%20%25%25sql#magics-e-g-sql-sql-etc)
+
+<!-- #region -->
+Here's an example of how to use Pandas and the JupySQL alternatives to transform data:
+```python
+# Rename columns
+df = data.rename(columns={'old_column_name': 'new_column_name'})  # Pandas
+%sql df << SELECT *, old_column_name AS new_column_name FROM data;  # JupySQL
```

Comment:
these SQL queries are hard to read since they take a single line, I think let's break it into multiple cells